Added silent mode and native i3 support

ororktn opened the pull request:
- `tuilade` won't require `i3-save-tree` as an input anymore and can now be directly used with `i3msg -t get_tree`
- Added silent mode (`-s`): will hide non needed information
- Added no-swallows mode (`-n`): will hide swallow nodes
- Added beginning of full tree printing (`-e`)
- Added whether to print parents of the beginning of the full tree print (`-p`)

Commit:
Added Clap with a first option: silent mode

Commit:
Started to implement silent mode

Commit:
🐛 (m) Replaced a `String::new` by a `format`

Commit:
Working on using directly i3-msg

- Added documentation

Commit:
Added --no-swallows option

Commit:
🐛 fixed a split_at that could split in the middle of a character

Commit:
Added display of focused tree

Commit:
✨ Added tree exploration

- Added a CLI option to select a level from which to print full tree of
  focused node
- Added a CLI option to print parents from root to beginning of
  expansion

Lymkwi commented on src/main.rs:
```diff
@@ -386,33 +516,42 @@ impl Node {
             },
-            swallows = if self.swallows.is_empty() {
+            swallows = if self.swallows.is_empty() || settings.no_swallows {
                 "No swallows"
```
Perhaps the whole swallows window could be hidden if there are none/they are disabled?

Lymkwi replied:
(this one's not necessary for merging, I'm just throwing the idea here)

Lymkwi commented on src/main.rs:
```diff
@@ -355,20 +426,79 @@ impl Node {
+        let print_self = print_children || settings.print_parents;
+
+        // Children
+        let children: Vec<&Node> = if print_children {
```
Use `Self` instead of `Node` for the `&` type.

Lymkwi commented on src/main.rs:
```diff
@@ -427,18 +566,28 @@ fn read_input() -> Result<String, String> {
+            mp.pretty_print(
+                &format!("{root_id}"),
+                &settings,
+                &mp.tree_type == &settings.expand_from
```
no need to take the references of both fields to compare them

Lymkwi reviewed (approved):
Nothing blocking, just a couple of things you could tweak here and there before I merge

Commit:
🤏 Small tweaks here and there

- Hide swallows when there are none (even when not in silent mode)
- Fixed an `&Node` to a `&Self` in `Node`
- Comparing fields instead of references
